src/ui/edit/missing-song-elements-dialog.c

```diff
@@ -77,8 +77,11 @@ make_listview (GtkWidget * vbox, GList * missing_elements, const gchar * msg)
     length += 2 + strlen ((gchar *) (node->data));
   }
   ptr = missing_text = g_malloc (length);
+  gint length_total = length;
   for (node = missing_elements; node; node = g_list_next (node)) {
     length = g_sprintf (ptr, "%s\n", (gchar *) (node->data));
+	length_total -= length;
+	g_assert(length_total >= 0);
```

Comment:
the g_assert will fail the program without much context for the user. The code before calculates the length. It would probably a good idea to extract the code into a helper and add a test, which I can take a look at. Please revert this part for now.

Reply:
Yes, just being paranoid. This is not like the other snprintf changes; it's in the class something that fires in the case of pure programmer errors (i.e. mistake during refactor) which is why I chose to put an assert there. I'll revert.